sarif/include/gap/sarif/definitions.hpp

```diff
@@ -0,0 +1,2866 @@
+// Copyright 2024-present, Trail of Bits, Inc.
+
+#pragma once
+
+#include <cstdint>
+#include <forward_list>
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+#include <nlohmann/json.hpp>
+
```

Comment:
Add some documentation, what was the source for these definitions, naming scheme etc. so one can easily relate to the specification.

Reply:
Done!